### PR TITLE
feat: add image and file upload

### PR DESCRIPTION
**Pull request by peterdeme**
Add image and file upload

### CODE REVIEW

File: pkg/cmd/chat/file/file.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+			cmd.Println("Successfully deleted image")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("channel-type", "t", "", "[required] Channel type to interact with")
+	fl.StringP("channel-id", "i", "", "[required] Channel id to interact with")
+	fl.StringP("image-url", "u", "", "[required] Url of the image to delete")
```
Comment (yaziine) [inline pkg/cmd/chat/file/file.go:187]:
```suggestion
	fl.StringP("image-url", "u", "", "[required] URL of the image to delete")
```

File: pkg/cmd/chat/utils/fileupload.go [hunk 2]
```diff
@@ -8,22 +8,56 @@ import (
...
-func UploadFile(c *stream.Client, cmd *cobra.Command, chType, chId, userID, filePath string) (string, error) {
+type uploadType string
+
+const (
+	uploadTypeFile  uploadType = "file"
+	uploadTypeImage uploadType = "image"
+)
+
+func UploadFile(c *stream.Client, cmd *cobra.Command, chType, chId, userID, filePath, contentType string) (string, error) {
+	return uploadFileInternal(c, cmd, uploadTypeFile, chType, chId, userID, filePath, contentType)
```
Comment (yaziine) [inline pkg/cmd/chat/utils/fileupload.go:19]: I don't think we need `Internal` suffix

File: pkg/cmd/chat/file/file.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chType, _ := cmd.Flags().GetString("channel-type")
+			chId, _ := cmd.Flags().GetString("channel-id")
+			imageUrl, _ := cmd.Flags().GetString("image-url")
+
+			err = utils.DeleteImage(c, cmd, chType, chId, imageUrl)
```
Comment (yaziine) [inline pkg/cmd/chat/file/file.go:174]: Why don't we call `c.Channel()...` directly here?

Comment (yaziine) [reply pkg/cmd/chat/file/file.go:174]: same concerns for delete file

File: pkg/cmd/chat/channel/channel_test.go [hunk 4]
```diff
@@ -111,5 +111,5 @@ func TestListChannel(t *testing.T) {
 	cmd.SetArgs([]string{"list-channels", "-t", "messaging", "-l", "1"})
 	_, err := cmd.ExecuteC()
 	require.NoError(t, err)
-	require.Contains(t, cmd.OutOrStdout().(*bytes.Buffer).String(), "messaging")
+	require.Contains(t, cmd.OutOrStdout().(*bytes.Buffer).String(), "Channels")
```
Comment (yaziine) [inline pkg/cmd/chat/channel/channel_test.go:114]: This test doesn't cover if the rendering is broken right? Shouldn't we add more tests for it?

File: pkg/cmd/chat/utils/fileupload.go (hunk 2 above)
Comment (peterdeme) [reply pkg/cmd/chat/utils/fileupload.go:19]: oh yeah, golang won't complain about it. fixed. ✅

File: pkg/cmd/chat/file/file.go (hunk 3 above)
Comment (peterdeme) [reply pkg/cmd/chat/file/file.go:174]: true, does not make much sense.

File: pkg/cmd/chat/channel/channel_test.go (hunk 4 above)
Comment (peterdeme) [reply pkg/cmd/chat/channel/channel_test.go]: Fixed the test

File: pkg/cmd/chat/channel/channel_test.go [hunk 5]
```diff
@@ -108,8 +108,14 @@ func TestUpdateChannelPartial(t *testing.T) {
 
 func TestListChannel(t *testing.T) {
 	cmd := test.GetRootCmdWithSubCommands(NewCmds()...)
+	chName := test.InitChannel(t)
+	t.Cleanup(func() {
+		test.DeleteChannel(chName)
+
```
Comment (ferhatelmas) [inline pkg/cmd/chat/channel/channel_test.go:114]: rm

File: pkg/cmd/chat/file/file.go [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+func uploadFileCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upload-file --channel-type [channel-type] --channel-id [channel-id] --user-id [user-id] --file [file]",
+		Short: "Upload a file",
+		Long: heredoc.Doc(`
+			Stream will not block any file types from uploading, however, different
+			clients may handle different types differently or not at all.
+			You can set a more restrictive list for your application if needed.
+			The maximum file size is 20MB.
```
Comment (ferhatelmas) [inline pkg/cmd/chat/file/file.go:27]: this is 100 mb now

Comment (ferhatelmas) [reply pkg/cmd/chat/file/file.go:27]: if copied from SDK docs or so, we need to update there too

File: pkg/cmd/chat/utils/fileupload.go [hunk 7]
```diff
@@ -8,19 +8,43 @@ import (
...
 	}
 	defer file.Close()
 
 	req := stream.SendFileRequest{
-		User:     &stream.User{ID: userID},
-		FileName: filepath.Base(file.Name()),
-		Reader:   file,
+		User:        &stream.User{ID: userID},
+		FileName:    filepath.Base(file.Name()),
+		ContentType: contentType,
```
Comment (ferhatelmas) [inline pkg/cmd/chat/utils/fileupload.go:36]: this isn't needed actually since backend does auto detection, will be better to drop flags for simplification

**Review by ferhatelmas — CHANGES_REQUESTED** [review]
two points

* size limit 100mb
* no need for content type

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**